src/ring_theory/noetherian.lean

```diff
@@ -578,6 +578,11 @@ class is_noetherian_ring (R) [ring R] extends is_noetherian R R : Prop
 theorem is_noetherian_ring_iff {R} [ring R] : is_noetherian_ring R ↔ is_noetherian R R :=
 ⟨λ h, h.1, @is_noetherian_ring.mk _ _⟩
 
+/-- A commutative ring is Noetherian if and only if all its ideals are finitely-generated. -/
+lemma is_noetherian_ring_iff_ideal_fg (R : Type*) [comm_ring R] :
+  is_noetherian_ring R ↔ ∀ I : ideal R, I.fg :=
+is_noetherian_ring_iff.trans ⟨λ h, h.noetherian, is_noetherian.mk⟩
```

Comment:
I think the `⟨λ h, h.noetherian, is_noetherian.mk⟩` part of this proof is probably worth extracting to a new lemma `is_noetherian_iff_submodule_fg` or even `is_noetherian_def`.

Reply:
I added `is_noetherian_def`